server/datastore/mysql/hosts_test.go

```diff
@@ -1815,6 +1815,24 @@ func testHostsTotalAndUnseenSince(t *testing.T, ds *Datastore) {
 	require.NoError(t, err)
 	assert.Equal(t, 3, total)
 	assert.Equal(t, 2, unseen)
+
+	// host not counted as unseen if less than a full 24 hours has passed
+	_, err = ds.writer.ExecContext(context.Background(), `UPDATE host_seen_times SET seen_time = ? WHERE host_id = 2`, time.Now().Add(-1*time.Duration(1)*23*time.Hour))
+	require.NoError(t, err)
+
+	total, unseen, err = ds.TotalAndUnseenHostsSince(context.Background(), 1)
+	require.NoError(t, err)
+	assert.Equal(t, 3, total)
+	assert.Equal(t, 1, unseen)
+
+	// host counted as unseen if more than 24 hours has passed
+	_, err = ds.writer.ExecContext(context.Background(), `UPDATE host_seen_times SET seen_time = ? WHERE host_id = 2`, time.Now().Add(-1*time.Duration(1)*25*time.Hour))
+	require.NoError(t, err)
+
+	total, unseen, err = ds.TotalAndUnseenHostsSince(context.Background(), 1)
+	require.NoError(t, err)
+	assert.Equal(t, 3, total)
+	assert.Equal(t, 2, unseen)
```

Comment:
So the bug was that we had to wait until the next day for it to count as unseen instead of right after, am I understanding correctly?

Reply:
No it was overcounting unseen hosts

Reply:
Oh, because this:

```
select datediff(current_date, 20220524194006);
```

returns 0, which means that in the above query it wouldn't have been >= than the interval, even though technically last seen at would be today-ish.

I guess in that case, going all the way down to seconds sounds like too much, but I suppose that's more of a nit than anything else.